File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BugDetails.swift
//
//  BugDetailCells.swift
//  bugTrap
//
//  Created by Colby L Williams on 8/23/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import Foundation
import CoreGraphics

struct BugDetails {
	
    enum DetailType {
		
		case Images
		case Selection		(String, String)
		case TextField		(String, String)
		case TextView		(String, String)
		case DateDisplay	(String, String)
		case DatePicker		(NSDate, String)
		
		
		func cellId() -> String {
			switch self {
			case .Images:		return "BtEmbeddedImageCollectionViewTableViewCell"
			case .Selection:	return "BtTitlePlaceholderTableViewCell"
			case .TextField:	return "BtTextFieldTableViewCell"
			case .TextView:		return "BtTextViewTableViewCell"
			case .DateDisplay:	return "BtDateDisplayTableViewCell"
			case .DatePicker:	return "BtDatePickerTableViewCell"
			}
		}
		
		
		func cellHeight() -> CGFloat {
			switch self {
			case .Images:		return 142.0
			case .Selection:	return 44.0
			case .TextField:	return 44.0
			case .TextView:		return 100.0
			case .DateDisplay:	return 44.0
			case .DatePicker:	return 163.0
			}
		}
		
		
		func isDate() -> Bool {
			switch self {
			case .Images, .Selection, .TextField, .TextView:
				return false
			case .DateDisplay, .DatePicker:
				return true
			}
		}
	}
}

File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Trackers/Common/Domain/BugDetails.cs
using System;

namespace bugTrapKit
{
	public class BugDetails
	{
		public enum DetailType
		{
			Images,
			Selection,
			TextField,
			TextView,
			DateDisplay,
			DatePicker
		}
	}

	public static class BugDetailsExtensions
	{

		public static string CellId (this BugDetails.DetailType detailType)
		{
			switch (detailType) {
			case BugDetails.DetailType.Images:
				return "BtEmbeddedImageCollectionViewTableViewCell";
			case BugDetails.DetailType.Selection:
				return "BtTitlePlaceholderTableViewCell";
			case BugDetails.DetailType.TextField:
				return "BtTextFieldTableViewCell";
			case BugDetails.DetailType.TextView:
				return "BtTextViewTableViewCell";
			case BugDetails.DetailType.DateDisplay:
				return "BtDateDisplayTableViewCell";
			case BugDetails.DetailType.DatePicker:
				return "BtDatePickerTableViewCell";
			default:
				return null;
			}
		}

		public static nfloat CellHeight (this BugDetails.DetailType detailType)
		{
			switch (detailType) {
			case BugDetails.DetailType.Images:
				return 142;
			case BugDetails.DetailType.Selection:
				return 44;
			case BugDetails.DetailType.TextField:
				return 44;
			case BugDetails.DetailType.TextView:
				return 100;
			case BugDetails.DetailType.DateDisplay:
				return 44;
			case BugDetails.DetailType.DatePicker:
				return 163;
			default:
				return 44;
			}
		}

		public static bool IsDate (this BugDetails.DetailType detailType)
		{
			return detailType == BugDetails.DetailType.DateDisplay || detailType == BugDetails.DetailType.DatePicker;
		}
	}
}